CREATE: New document for documentation process

sksingh2022 opened the pull request:
# Description

Updating documentation for the documentation.md file. Added content outlining the documentation processes and the information corresponds with the presentation by Talia on documentation ups killing

Fixes # (issue)

No fixes. Just adding content into the company handbook

- [x] Documentation (new)

# How Has This Been Tested?

Not applicable, this is a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

sksingh2022 committed:
CREATE: New document for documentation process

ghost commented:
What is this document for? 
The PR template has not been filled out?

sksingh2022 committed:
ADD: File path to documentation .md

ghost commented:
I would consider making this a draft PR, and add content before you merge the blank file.

ghost commented:
@sksingh2022, there are now two PRs which will likely create a merge conflict? Would you like to close one of the PRs?

sksingh2022 commented:
Hi talia, sorry about this, please see updated.

sksingh2022 committed:
UPDATE: Documentation.md

Added content onto the markdown file.

sksingh2022 committed:
UPDATE: Documentation.md content

sksingh2022 committed:
UPDATE: Content updated on documentation.md

sksingh2022 committed:
Update: documentation.md

sksingh2022 committed:
update: documentation.md content

sksingh2022 committed:
update: documentation.md content

Commit:
fix: run npm

Npm was not adding changes to @sksingh2022 environment. Have uploaded on behalf of him

Commit:
Merge branch 'updatehandbook' of https://github.com/thoth-tech/handbook into updatehandbook

Commit:
fix: removing line

sksingh2022 committed:
UPDATE: DOCUMENTATION.MD

sksingh2022 committed:
update: hyoerlinks within page.

sksingh2022 committed:
Merge branch 'updatehandbook' of https://github.com/thoth-tech/handbook into updatehandbook

sksingh2022 committed:
Update: hyperlinks on page #2

sksingh2022 committed:
Update README.md

Co-authored-by: talzid <101184078+talzid@users.noreply.github.com>

sksingh2022 committed:
fix: update indentation

sksingh2022 committed:
fix: npm run

sksingh2022 committed:
fix: formatting issues

sksingh2022 committed:
update: redundant information and updating links

talzid commented on docs/processes/documentation/documentation.md:
```diff
@@ -0,0 +1,37 @@
+## Introduction
+
+Documentation is one of the most important parts of any software epic. It is imperative to remember, you will not be working on the capstone projects forever and that there will be people taking your place within Thoth Tech.
```
I would prefer if our handbook was less capstone related and more company focused.
Although, Thith Tech is for capstone, it is still a viable open source company and should be spoken about as such.

Consider rephrasing: 

```suggestion
Thoth Tech prides itself on our ability to produce sustainable solutions, this can be achieved through documentation.
Documentation is an important element in the software development life cycle. It continues throughout the entire project.
It is imperative to remember, others will take over your projects or want to learn more about how it works and why. 
```

talzid commented on docs/processes/documentation/documentation.md:
```diff
@@ -0,0 +1,37 @@
+
+Documentation is one of the most important parts of any software epic. It is imperative to remember, you will not be working on the capstone projects forever and that there will be people taking your place within Thoth Tech.
+
+Documentation is the offical information that is provided by the company for users and contributors. Documentation should provide clarity around troubleshooting and provide users and those involved with a basic idea of what it is the platform does and how it works.
```
Hopefully it is more than a basic understanding 


```suggestion
Documentation is offical information, provided by the company for users and contributors. Documentation is written for specific audiences and provide knowledge on the platform.
```

talzid commented on docs/processes/documentation/documentation.md:
```diff
@@ -0,0 +1,37 @@
+
+## Why is it important?
+
+Without documentation, there is no record of:
```
Not entirely true, it is possible to find a record but it is difficult.

I would remove this section and the following list. Easier to link to find information in one spot such as the Docs repo.

sksingh2022 replied:
Section removed

talzid commented on docs/processes/documentation/documentation.md:
```diff
@@ -0,0 +1,37 @@
+- Consistent
+- Accessible
+
+Further to this, we ask you consider reading the documentation upskilling guide on the [training centre](docs/../../../learning/training/writing-documentation.md)
```
As this is essentially an index page that might have future links, I would consider a list. 


```suggestion
For more information:

- [writing documentation guide](docs/../../../learning/training/writing-documentation.md)
```

talzid commented on docs/processes/documentation/documentation.md:
```diff
@@ -0,0 +1,37 @@
+
+## Contributing
+
+Please see the [contribution guidelines](docs/processes/../../../quality-assurance/git-contribution-guide.md)
```
This should be linked to in documentation repo readme.md. 

As it is already mentioned in the handbook on the readme.md I would refrain from linking to it again.

sksingh2022 replied:
Section also removed - thank you

talzid commented on docs/processes/documentation/documentation.md:
```diff
@@ -0,0 +1,37 @@
+- What project updates have been added
+- And more
+
+Thoth Tech documentation are not only used by the one respective team. The documentation can be accesed at anytime by any of the Thoth Tech company members just to see where a project is heading and what progress has been made. Additionally, should other teams wish, they may collaborate and think of intuitve ways in which to further improve the projects.
```
I personally would not include the section about collaboration, I do not think this is the right place to suggest such activity and will depend on the area leads of that tri. Better to leave that go future area leads and have one source of truth.

```suggestion
Thoth Tech documentation is accessible to all.
```

talzid reviewed (changes requested):
Thank you @sksingh2022 for your initiative creating this document. I have left some suggestions for you to consider that might help with the flow between other documents in the handbook.
Great job. Would be nice to see a similar message in the Docs repo readme.md

sksingh2022 committed:
Update docs/processes/documentation/documentation.md

Co-authored-by: talzid <101184078+talzid@users.noreply.github.com>

sksingh2022 committed:
Update docs/processes/documentation/documentation.md

Co-authored-by: talzid <101184078+talzid@users.noreply.github.com>

sksingh2022 committed:
Update docs/processes/documentation/documentation.md

Co-authored-by: talzid <101184078+talzid@users.noreply.github.com>

sksingh2022 committed:
Update docs/processes/documentation/documentation.md

Co-authored-by: talzid <101184078+talzid@users.noreply.github.com>

sksingh2022 committed:
UPDATE: RESOLVED COMMENTS BY TALIA

sksingh2022 committed:
Merge branch 'main' into updatehandbook

sksingh2022 commented:
Hi Talia, I have successfully updated the markdown file and implemented your suggestions. Thank you for your support.

ghost commented:
@tancnle is there a way to only merge some files and not others?
This PR has added unwanted changes to the package file and prettier file.

tancnle commented:
> @tancnle is there a way to only merge some files and not others? This PR has added unwanted changes to the package file and prettier file.

@talzid We have to check out this branch, remove the unnecessary changes and do another commit. I can do that if there is no objection.

ghost commented:
> > @tancnle is there a way to only merge some files and not others? This PR has added unwanted changes to the package file and prettier file.
> 
> @talzid We have to check out this branch, remove the unnecessary changes and do another commit. I can do that if there is no objection.

Please do! Could you also explain why this may have happened ?

tancnle commented:
> Please do! Could you also explain why this may have happened?

The change might be made manually or it was a result of an unclean merge from the `main` branch. Looking at the history, it seems like the latter.

tancnle commented on docs/processes/documentation/documentation.md:
```diff
@@ -0,0 +1,26 @@
+## Introduction
```
**question:** @talzid @sksingh2022 What is the difference between this doc and the [one](https://github.com/thoth-tech/handbook/blob//docs/documentation/index.md) we currenttly have in this folder?

tancnle commented on README.md:
```diff
@@ -31,7 +31,7 @@ Please see the [contribution guidelines](CONTRIBUTING.md).
   - Developer Operations
   - [Quality Assurance](docs/processes/quality-assurance/quality-assurance-overview.md)
-  - [Documentation](docs/documentation/index.md)
+  - [Documentation](docs/processes/documentation/documentation.md)
```
**issue:** Is this change intended? We already had the file `docs/documentation/index.md` in the `main` branch. This change will effectively remove its reference from this file.

talzid replied:
As per above, I was going to merge the files into one once merged and keep the index name for consistency. But did not want to 'rob' Shivam of the commit evidence

tancnle replied:
> But did not want to 'rob' Shivam of the commit evidence

Gotcha. It is not really robing by the way. Shivam's change is still presented in this PR. We don't want to merge changes that are conflicting or duplicating.

By the way, I have removed the prettier and package changes in this PR. Please proceed with updating the documentation references and content whenever you see fit.

tancnle committed:
fix: remove unintended change to packages

Commit:
fix: files merged and consistent

talzid reviewed (commented):
Changes made and file merge confirmed

sksingh2022 commented:
Thank you @tancnle & @talzid for your help!
